### PR TITLE
Test before pushing your image (docs)

### LINKED ISSUE #269: How can I integrate with container-structure-test?
Most likely it would require a new input, such as:

`pre-publish-command: container-structure-test --image test/image`

### PR DESCRIPTION
**Pull request by crazy-max**
Fixes #269

cc @felipecrs @corey-dawson @francisdb

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: docs/advanced/test-before-push.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        name: Build and export to Docker
+        with:
```
Comment (francisdb) [inline docs/advanced/test-before-push.md:43]: no uses / run?

Comment (crazy-max) [reply docs/advanced/test-before-push.md:43]: Thanks indeed!

Comment (crazy-max) [reply docs/advanced/test-before-push.md:44]: Done

File: docs/advanced/test-before-push.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        name: Build and export to Docker
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          load: true
+          tags: ${{ env.LOCAL_TAG }}
+      -
+        name: Test
+        run: |
+          docker run --rm ${{ env.TEST_TAG }}
```
Comment (francisdb) [inline docs/advanced/test-before-push.md:51]: what's the difference between `LOCAL_TAG` and `TEST_TAG`?

Comment (crazy-max) [reply docs/advanced/test-before-push.md:51]: Damn another typo... Fixed thanks again for the review!

**Comment by francisdb** [comment]
So this should work with the currently released versions?

**Comment by crazy-max** [comment]
> So this should work with the currently released versions?

Yes

File: docs/advanced/test-before-push.md (hunk 2 above)
Comment (francisdb) [reply docs/advanced/test-before-push.md:51]: you're welcome, thanks for documenting this

**Comment by felipecrs** [comment]
Thanks @crazy-max for documenting this. I was doing the same thing here; the approach is reasonable.

**Comment by crazy-max** [comment]
@felipecrs Thanks for your feedback!

**Pull request merged**